awscli/testutils.py

```diff
@@ -775,6 +775,16 @@ def head_object(self, bucket_name, key_name):
         response = client.head_object(Bucket=bucket_name, Key=key_name)
         return response
 
+    def wait_until_key_exists(self, bucket_name, key_name, extra_params=None,
+                              min_successes=3):
+        client = self.create_client_for_bucket(bucket_name)
+        waiter = client.get_waiter('object_exists')
+        params = {'Bucket': bucket_name, 'Key': key_name}
+        if extra_params is not None:
+            params.update(extra_params)
+        for _ in range(min_successes):
```

Comment:
Possible? yes.  Likely? no.

I'm ok with trading off a second or two in runtime to ensure the rare cases are even less likely to happen.